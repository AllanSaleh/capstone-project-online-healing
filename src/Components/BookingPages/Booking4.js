import React from 'react';

export default function Booking4() {
  const choices = [
    {
      id: 1,
      content: 'I prefer a Male counselor',
    },
    {
      id: 2,
      content: 'I prefer a Female counselor',
    },
    {
      id: 3,
      content: 'I prefer an Older(45+) counselor',
    },
    {
      id: 4,
      content: 'I prefer a Non-Religious counselor',
    },
    {
      id: 5,
      content: 'I prefer a Younger(30-) counselor',
    },
  ];

  return (
    <div className="h-firstsection px-sides pt-navbar">
      <h1 className="text-title uppercase mb-4">
        {' '}
        letâ€™s match you with the right therapist
      </h1>
      <h3 className="text-subtitle opacity-50 text-justify">
        Please fill out this short questionnaire to provide some general and
        anonymous background about you and the issues you'd like to deal with in
        online therapy. It would help us match you with the most suitable
        therapist for you.
      </h3>

      <div className="flex flex-col justify-evenly h-2/3 max-w-3xl my-16 mx-auto px-8 py-4 shadow-md">
        <div>
          Are there any specific quality that you would prefer in a counselor?
        </div>

        {choices.map((choice) => (
          <div>
            <input
              type="radio"
              id={choice.id}
              name="counselor_options"
              value={choice.content}
            />
            <label className="ml-2" htmlFor={choice.id}>
              {choice.content}
            </label>
          </div>
        ))}

        <div className="flex justify-between">
          <button
            type="button"
            className="uppercase bg-blue-dark h-12 w-32 text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
          >
            back
          </button>
          <button
            type="button"
            className="uppercase bg-blue-dark h-12 w-32 text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
          >
            next
          </button>
        </div>
      </div>
    </div>
  );
}
