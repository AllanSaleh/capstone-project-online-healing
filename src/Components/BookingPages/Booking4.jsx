import React from 'react';
import { useHistory } from 'react-router-dom';

export default function Booking4() {
  const choices = [
    {
      id: 1,
      content: 'I prefer a Male counselor',
    },
    {
      id: 2,
      content: 'I prefer a Female counselor',
    },
    {
      id: 3,
      content: 'I prefer an Older(45+) counselor',
    },
    {
      id: 4,
      content: 'I prefer a Non-Religious counselor',
    },
    {
      id: 5,
      content: 'I prefer a Younger(30-) counselor',
    },
  ];

  const history = useHistory();

  const PrevPage = () => {
    // Put firestore code here!
    window.scrollTo(0, 0);
    history.push('/Booking3');
  };
  const NextPage = () => {
    // Put firestore code here!
    window.scrollTo(0, 0);
    history.push('/Booking5');
  };

  return (
    <div className="h-firstsection px-sides pt-navbar">
      <h1 className="text-3xl lg:text-title uppercase my-8">
        letâ€™s match you with the right therapist
      </h1>
      <h3 className="text-md lg:text-subtitle opacity-50 text-justify">
        Please fill out this short questionnaire to provide some general and anonymous background
        about you and the issues you'd like to deal with in online therapy. It would help us match
        you with the most suitable therapist for you.
      </h3>

      <div className="flex flex-col justify-evenly w-full md:max-w-md lg:max-w-2xl my-16 mx-auto px-8 py-4 shadow-md">
        <div className="text-md lg:text-subtitle">
          Are there any specific quality that you would prefer in a counselor?
        </div>

        <div className="flex flex-col justify-around my-8">
          {choices.map((choice) => (
            <div className="flex items-center my-2">
              <input type="radio" id={choice.id} name="counselor_options" value={choice.content} />
              <label className="text-sm lg:text-paragraph ml-2" htmlFor={choice.id}>
                {choice.content}
              </label>
            </div>
          ))}
        </div>
        <div className="flex justify-between">
          <button
            onClick={() => PrevPage()}
            type="button"
            className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
          >
            back
          </button>
          <button
            onClick={() => NextPage()}
            type="button"
            className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition-all duration-300"
          >
            next
          </button>
        </div>
      </div>
    </div>
  );
}
