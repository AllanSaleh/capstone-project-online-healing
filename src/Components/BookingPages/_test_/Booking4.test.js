import * as React from 'react';
import Booking4 from "../Booking4";
import * as renderer from "react-test-renderer";

it('Booking4 renders correctly', () => {
  const tree = renderer
    .create(<Booking4/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});