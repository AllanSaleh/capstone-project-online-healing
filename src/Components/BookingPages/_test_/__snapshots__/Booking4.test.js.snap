// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking4 renders correctly 1`] = `
<div
  className="px-sides pt-navbar"
>
  <h1
    className="text-3xl lg:text-title uppercase mb-4"
  >
    letâ€™s match you with the right therapist
  </h1>
  <h3
    className="text-md lg:text-subtitle opacity-50 text-justify"
  >
    Please fill out this short questionnaire to provide some general and anonymous background about you and the issues you'd like to deal with in online therapy. It would help us match you with the most suitable therapist for you.
  </h3>
  <div
    className="flex flex-col justify-evenly w-full md:max-w-md lg:max-w-2xl my-16 mx-auto px-8 py-4 shadow-md"
  >
    <div
      className="text-md lg:text-subtitle"
    >
      Are there any specific quality that you would prefer in a counselor?
    </div>
    <div
      className="flex flex-col justify-around my-8"
    >
      <div
        className="flex items-center"
      >
        <input
          id={1}
          name="counselor_options"
          type="radio"
          value="I prefer a Male counselor"
        />
        <label
          className="text-sm lg:text-paragraph ml-2"
          htmlFor={1}
        >
          I prefer a Male counselor
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={2}
          name="counselor_options"
          type="radio"
          value="I prefer a Female counselor"
        />
        <label
          className="text-sm lg:text-paragraph ml-2"
          htmlFor={2}
        >
          I prefer a Female counselor
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={3}
          name="counselor_options"
          type="radio"
          value="I prefer an Older(45+) counselor"
        />
        <label
          className="text-sm lg:text-paragraph ml-2"
          htmlFor={3}
        >
          I prefer an Older(45+) counselor
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={4}
          name="counselor_options"
          type="radio"
          value="I prefer a Non-Religious counselor"
        />
        <label
          className="text-sm lg:text-paragraph ml-2"
          htmlFor={4}
        >
          I prefer a Non-Religious counselor
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={5}
          name="counselor_options"
          type="radio"
          value="I prefer a Younger(30-) counselor"
        />
        <label
          className="text-sm lg:text-paragraph ml-2"
          htmlFor={5}
        >
          I prefer a Younger(30-) counselor
        </label>
      </div>
    </div>
    <div
      className="flex justify-between"
    >
      <button
        className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
        type="button"
      >
        back
      </button>
      <button
        className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
        type="button"
      >
        next
      </button>
    </div>
  </div>
</div>
`;
